### PR TITLE
[istio] Improved resourceManagement validation in the ModuleConfig

### PR DESCRIPTION
**Pull request by skurbatov**
## Description

Improved validation of the `spec.sidecar.resourcesManagement.static` parameter in istio ModuleConfig.

## Why do we need it, and what problem does it solve?

Onder-the-hood resource `IstioOperator` doesn't support the numberic resource specifications, it can handle only the string's. We decided to handle it by forbidding the numbering notation in ModuleConfig.

## What is the expected result?

The values that you enter in `moduleConfig` do not cause errors in the Istio operation. If you enter an invalid value, you will immediately know about it before it is applied.

## Why do we need it in patch release?

The consequences of wrong configuration aren't obvious, istio-operator just stops working and the logs aren't obvious.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: istio
type: fix
summary: Improved validation of the ModuleConfig
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/110-istio/openapi/config-values.yaml [hunk 1]
```diff
@@ -299,10 +299,10 @@ properties:
                   Resource requests settings for pods.
                 properties:
                   cpu:
+                    x-examples: ["1", "1000m"]
                     oneOf:
                       - type: string
-                        pattern: "^[0-9]+m?$"
-                      - type: number
+                        pattern: '^(["].*["]|[0-9]+m)$'
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/openapi/config-values.yaml:305]:
```suggestion
                        pattern: '^(["][0-9]+["]|[0-9]+m)$'
```

**Review by AndreyPavlovFlant — CHANGES_REQUESTED** [review]

File: modules/110-istio/openapi/doc-ru-config-values.yaml [hunk 2]
```diff
@@ -150,7 +150,7 @@ properties:
                 properties:
                   cpu:
                     description: |
-                      Настройка запроса CPU (CPU requests).
+                      Настройка запроса CPU (CPU requests). Пример указания параметра - "1", "1000m".
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/openapi/doc-ru-config-values.yaml:153]:
```suggestion
                      Настройка запроса CPU (CPU requests).
```

**Review by AndreyPavlovFlant — CHANGES_REQUESTED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by tarabrind — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by AndreyPavlovFlant — APPROVED** [review]

**Review by tarabrind — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [10049](https://github.com/deckhouse/deckhouse/pull/10049) to the branch [release-1.64](https://github.com/deckhouse/deckhouse/tree/release-1.64) successful!

**Pull request merged**